pkg/sql/colexec/colexecbase/cast_tmpl.go

```diff
@@ -85,19 +83,28 @@ func GetCastOperator(
 			outputIdx:                resultIdx,
 		}, nil
 	}
-	leftType, rightType := fromType, toType
-	switch typeconv.TypeFamilyToCanonicalTypeFamily(leftType.Family()) {
-	// {{range .LeftFamilies}}
-	case _LEFT_CANONICAL_TYPE_FAMILY:
-		switch leftType.Width() {
-		// {{range .LeftWidths}}
-		case _LEFT_TYPE_WIDTH:
-			switch typeconv.TypeFamilyToCanonicalTypeFamily(rightType.Family()) {
-			// {{range .RightFamilies}}
-			case _RIGHT_CANONICAL_TYPE_FAMILY:
-				switch rightType.Width() {
-				// {{range .RightWidths}}
-				case _RIGHT_TYPE_WIDTH:
+	if toType.Identical(fromType) {
+		// We have an identity cast, so we use a custom identity cast operator.
+		return &castIdentityOp{
+			OneInputInitCloserHelper: colexecop.MakeOneInputInitCloserHelper(input),
```

Comment:
[nit] Since the last commit is for cleanup anyway, it might be nice to pull the common fields out into an embedded struct. What do you think?